File: sdk/nodejs/examples/streaming.ts
import { AgentKernel } from "agentkernel";

const client = new AgentKernel();

// Stream output from a command
for await (const event of client.runStream(["python3", "-c", "print('Hello from streaming!')"])) {
  switch (event.type) {
    case "started":
      console.log("[started]", event.data);
      break;
    case "output":
      process.stdout.write(String(event.data.data));
      break;
    case "done":
      console.log("\n[done] exit_code:", event.data.exit_code);
      break;
    case "error":
      console.error("[error]", event.data.message);
      break;
  }
}

File: sdk/python/examples/streaming.py
"""Streaming example for agentkernel Python SDK."""

import sys

from agentkernel import AgentKernel

client = AgentKernel()

# Stream output from a command
for event in client.run_stream(["python3", "-c", "print('Hello from streaming!')"]):
    match event.event_type:
        case "started":
            print("[started]", event.data)
        case "output":
            sys.stdout.write(event.data.get("data", ""))
        case "done":
            print(f"\n[done] exit_code: {event.data.get('exit_code')}")
        case "error":
            print(f"[error] {event.data.get('message')}", file=sys.stderr)
